src/Microsoft.AspNet.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcCoreBuilderExtensions.cs

```diff
@@ -136,6 +139,11 @@ internal static void AddViewServices(IServiceCollection services)
             // These are stateless so their lifetime isn't really important.
             services.TryAddSingleton<ITempDataDictionaryFactory, TempDataDictionaryFactory>();
             services.TryAddSingleton<SaveTempDataFilter>();
+
+            services.TryAddSingleton(ArrayPool<ViewBufferValue>.Create(MemoryPoolViewBufferScope.SegmentSize, 50));
```

Comment:
That's the default **maximum** that will be retained by the pool. You can ask for any size you want, the pool will create multiple sizes.